src/view/textproxy.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * TextProxy is a wrapper for substring of {@link engine.view.Text}. Instance of this class is created by
+ * {@link engine.view.TreeWalker} when only a part of {@link engine.view.Text} needs to be returned.
+ *
+ * **Note:** TextProxy instances are created on the fly basing on the current state of parent {@link engine.view.Text}.
+ * Because of this it is highly unrecommended to store references to TextProxy instances because they might get
+ * invalidated due to operations on Document. Also TextProxy is not a {@link engine.view.Node} so it can not be
+ * inserted as a child of {@link engine.view.Element}.
+ *
+ * You should never create an instance of this class by your own. Instead, use string literals or {@link engine.model.Text}.
+ *
+ * @memberOf engine.view
+ */
+export default class TextProxy {
+	/**
+	 * Creates a tree view text proxy.
+	 *
+	 * @param {engine.view.Text} textNode Text node which text proxy is a substring.
+	 * @param {Number} startOffset Offset from beginning of {#_textNodeParent} and first character of {#_data}.
+	 * @param {Number} [length] Length of substring. If is not set then the end offset is at the end of {#_textNodeParent}.
+	 */
+	constructor( textNode, startOffset, length ) {
+		/**
+		 * Element that is a parent of this text proxy.
+		 *
+		 * @readonly
+		 * @member {engine.view.Element|engine.view.DocumentFragment|null} engine.view.Node#parent
+		 */
+		this.parent = textNode.parent;
+
+		/**
+		 * Reference to the {@link engine.view.Text} element which TextProxy is a substring.
+		 *
+		 * @protected
+		 * @readonly
+		 * @member {engine.view.Text} engine.view.TextProxy#_textNode
+		 */
+		this._textNode = textNode;
```

Comment:
There should be a way to get a text node from the text proxy. This method should be public.